### PR TITLE
Add dev report for may 1st

### PR DESCRIPTION
**Pull request by vieux**
related to https://github.com/moby/moby/issues/32865

### CODE REVIEW

**Comment by tonistiigi** [comment]
@vieux Maybe separate the builder section into subdir and link it from the main one. Then it is possible to have some continuity when following a specific component and if a component gets split up the reports can move as well.

File: reports/2017-05-01.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+- [Refactor Dockerfile.parser and directive](https://github.com/moby/moby/pull/32580)
+- [Refactor builder dispatch state](https://github.com/moby/moby/pull/32600)
+- [Use a bytes.Buffer for shell_words string concat](https://github.com/moby/moby/pull/32601/files)
+- [Refactor `Builder.commit()`](https://github.com/moby/moby/pull/32772)
+- [Remove b.escapeToken, create ShellLex](https://github.com/moby/moby/pull/32858)
+
+#### New feature: Long running session
+
+PR for [adding long-running session between daemon and cli]() that enabled advanced features like incremental context send, build credentials from the client, ssh forwarding etc. is looking for initial design review. It is currently open if features implemented on top of it would use a specific transport implementation on the wire or a generic interface(current implementation). @tonistiigi is working on adding persistent cache capabilities that are currently missing from that PR. It also needs to be figured out how the [cli split](https://github.com/moby/moby/pull/32694) will affect features like this.
```
Comment (AkihiroSuda) [inline reports/2017-05-01.md]: Link broken

Comment (AkihiroSuda) [inline reports/2017-05-01.md]: ↑for "adding long-running session between daemon and cli"

Comment (vieux) [inline reports/2017-05-01.md]: thanks

**Comment by vieux** [comment]
@tonistiigi like this ?

**Review by vdemeester — APPROVED** [review]
LGTM 👍

**Comment by AkihiroSuda** [comment]
I think mixing up current `docker build`and future buildkit is confusing..

`docker build` should be documented in regular report?

**Comment by AkihiroSuda** [comment]
PTAL @tonistiigi

**Comment by tonistiigi** [comment]
> I think mixing up current docker buildand future buildkit is confusing..

Yes, I think we should call the subdir/component "builder" or "build". buildkit is just one of the topics in this week report. Otherwise SGTM

File: reports/2017-05-01.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Development Report for May 01, 2017
+
+This is the 1st report, since the Moby project was announced at DockerCon. Thank you to everyone that stayed a day late to attend the summit on Thursday.
```
Comment (tiborvass) [inline reports/2017-05-01.md]: `an extra day`

File: reports/builder/2017-05-01.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+To improve quality and performance, a new [proposal was made for switching the dependency interface](https://github.com/moby/moby/issues/32904) for current builder package. That should fix the current problems with data leakage and conflicts caused by daemon state cleanup scripts.
+
+@dnephin is in progress of refactoring current builder code to logical areas as a preparation work for updating this interface.
+
+Merged as part of this effort:
+
+- [Refactor Dockerfile.parser and directive](https://github.com/moby/moby/pull/32580)
+- [Refactor builder dispatch state](https://github.com/moby/moby/pull/32600)
+- [Use a bytes.Buffer for shell_words string concat](https://github.com/moby/moby/pull/32601/files)
```
Comment (AkihiroSuda) [inline reports/builder/2017-05-01.md]: nit: extra `/files`

File: reports/builder/2017-05-01.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+- [Refactor `Builder.commit()`](https://github.com/moby/moby/pull/32772)
+- [Remove b.escapeToken, create ShellLex](https://github.com/moby/moby/pull/32858)
+
+### New feature: Long running session
+
+PR for [adding long-running session between daemon and cli](https://github.com/moby/moby/pull/32677) that enabled advanced features like incremental context send, build credentials from the client, ssh forwarding etc. is looking for initial design review. It is currently open if features implemented on top of it would use a specific transport implementation on the wire or a generic interface(current implementation). @tonistiigi is working on adding persistent cache capabilities that are currently missing from that PR. It also needs to be figured out how the [cli split](https://github.com/moby/moby/pull/32694) will affect features like this. 
+
+### Proposals for new Dockerfile features that need design feedback:
+
+[Add IMPORT/EXPORT commands to Dockerfile](https://github.com/docker/docker/issues/32100)
```
Comment (AkihiroSuda) [inline reports/builder/2017-05-01.md]: nit: docker/docker -> moby/moby (although GitHub redirects it automatically)

**Review by thaJeztah — APPROVED** [review]
LGTM

minor nit: we should wrap these to 80-chars for consistency with other Markdown

File: reports/2017-05-01.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+### The CLI split
+
+Ongoing work to split the Docker CLI into [https://github.com/docker/cli](https://github.com/docker/cli) is happening [here](https://github.com/moby/moby/pull/32694).
+We are almost done, it should be merged soon.
+
+### Mailing list
+
+Slack works great for synchronous communication, but we need to place for async discussion. A mailing list is currently being setup.
+
+#### Find a good an non-confusing home for the remaining monolith
```
Comment (tiborvass) [inline reports/2017-05-01.md]: `s/an //`

File: reports/2017-05-01.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+Lots of discussion and progress made on this topic, see [here](https://github.com/moby/moby/issues/32871). The work will start this week.
+
+## Componentization
+
+So far only work on the builder happend regarding the componentization effort.
```
Comment (tiborvass) [inline reports/2017-05-01.md]: `happened`

**Review by tonistiigi — APPROVED** [review]
LGTM

**Comment by vieux** [comment]
@thaJeztah with all the links, it's tricky to wrap 80chars.

I took inspiration from https://github.com/containerd/containerd/blob/master/reports/2017-04-28.md

**Comment by thaJeztah** [comment]
get a better editor :wink:

**Pull request merged**